test/lib/pipeline.test.js

```diff
@@ -299,6 +305,32 @@ describe('Pipeline Model', () => {
         });
     });
 
+    describe('get secrets', () => {
+        it('has a secrets getter', () => {
+            const listConfig = {
+                params: {
+                    pipelineId: pipeline.id
+                },
+                paginate: {
+                    count: 25, // This limit is set by the matrix restriction
+                    page: 1
+                }
+            };
+
+            secretFactoryMock.list.resolves(null);
+            // when we fetch secrets it resolves to a promise
+            assert.isFunction(pipeline.secrets.then);
+            // and a factory is called to create that promise
+            assert.calledWith(secretFactoryMock.list, listConfig);
+
+            // When we call pipeline.secrets again it is still a promise
+            assert.isFunction(pipeline.secrets.then);
+            // ...but the factory was not recreated, since the promise is stored
+            // as the model's pipeline property, now
```

Comment:
Darren, you code should work, too. The original tests didn't really care about the output of the Promise, just that the getter returned something that looked like a promise. I agree with Dao, if we want to change it to your way, we should change all the getter tests at once and be consistent.